src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -60,13 +61,31 @@ public CommandResult executeUndoableCommand() throws CommandException {
         requireNonNull(model);
         try {
             model.addParcel(toAdd);
+            model.maintainSorted();
+            if (model.hasSelected()) {
+                ReadOnlyParcel previous = model.getAddressBook()
+                                               .getParcelList()
+                                               .get(model.getPrevIndex().getZeroBased());
+                if (previous.compareTo(toAdd) > 0) {
+                    model.forceSelect(model.getPrevIndex());
+                } else {
+                    model.forceSelect(Index.fromZeroBased(findIndex(previous)));
+                    System.out.println(model.getAddressBook().getParcelList().get(findIndex(previous)));
+                }
+            } else {
+                model.maintainSorted();
+            }
```

Comment:
Can you add comments for this snippet? I don't understand what is going on here.